app/javascript/controllers/subject_preference_form_controller.js

```diff
@@ -0,0 +1,30 @@
+import { Controller } from "stimulus"
+
+export default class extends Controller {
+  static targets = [ 'degreeSubjectContainer', 'degreeSubject', 'degreeStagesContainer' ]
+
+  connect() {
+    const radioButtons = this.degreeStagesContainerTarget.querySelectorAll('input[type="radio"]')
+    const checked = Array.from(radioButtons).find(r => r.checked)
+    if (checked) { this.toggleDegreeSubjectContainer(checked) }
+  }
+
+  degreeStageSelected(e) {
+    this.toggleDegreeSubjectContainer(e.target)
+  }
+
+  toggleDegreeSubjectContainer(radioButton) {
+    const requiresSubject = JSON.parse(radioButton.dataset.requiresSubject)
+    const options = this.degreeSubjectTarget.options
+    const valueForNoDegree = this.degreeSubjectTarget.dataset.valueForNoDegree
+    const optionForNoDegree = Array.from(options).find(o => o.value == valueForNoDegree)
+
+    this.degreeSubjectContainerTarget.hidden = !requiresSubject
+    optionForNoDegree.hidden = requiresSubject
+    if ( !requiresSubject ) {
```

Comment:
This looks fine, just one query. What's the advantage of setting `optionForNoDegree.selected = 'selected'` here rather than using boolean `true`?

```javascript
optionForNoDegree.selected = !requiresSubject
```

Reply:
When I manually select an option the option tag has the selected attribute set to 'selected' rather than true, so I thought I'd stick with that!

Reply:
Ah ok, fair enough. I _think_ anything truthy will work but it's all good 👍🏽